File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = [0, 0]

    while s != '1':
        new = ''
        for i in s:
            if i == '0':
                answer[1] += 1
            else:
                new += '1'

        bin_new = len(new)
        arr = []
        while bin_new>0:
            arr.append(str(bin_new % 2))
            bin_new //= 2

        arr.reverse()
        s = ''
        for i in arr:
            s += i

        answer[0] += 1

    return answer

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer(2,0);
    while(s.size()>1){
        //1단계
        int cnt = 0;
        string first = "";
        for(auto now : s){
            if(now=='0'){
                answer[1]++;
                continue;
            }
            first+=now;
        }
        //2단계
        int L = first.size();
        deque<int>q;
        while(L>0){
            q.push_back(L%2);
            L /= 2;
        }
        string second ="";
        while(!q.empty()){
            second += to_string(q.front());
            q.pop_front();
        }
        s = second;
        answer[0]++;
    }
    return answer;
}